docs/adr/0013-docker-latest-tag.md

```diff
@@ -0,0 +1,52 @@
+# 13) Docker latest tag {#adr_0013}
+
+Date: 2021-05-27
+
+## Status
+
+proposed
+
+## Context
+
+Docker allows tags to be defined for the images in a docker repository. The default tag that is used unless anything else is specified is the ["latest"](https://docs.docker.com/engine/reference/commandline/tag/#tag-an-image-referenced-by-name) tag. This goes both for publishing and pulling images, unless anything else is specified you get "latest".
+
+Previously the wikibase repository has shipped `wikibase/wikibase` images with three different latest tags depending on the flavor of wikibase.
+
+* latest-bundle
+* latest-base
+* latest (seems to be the base)
+
+As we now have a separate image called `wikibase/wikibase-bundle` the need for latest-bundle / latest-base is gone and we need to decide if tagging each image with "latest" is something we still want to do for each of the images we are to publish.
+
+There are many arguments being presented against using "latest" when running docker images[[1]](https://vsupalov.com/docker-latest-tag/)[[2]](https://blog.container-solutions.com/docker-latest-confusion). It comes with an uncertainty of what image we are actually getting, as latest does not necessarily mean the last image uploaded or the latest version but rather just the image tagged as "latest". It sort of serves as an easy and rather uninformed way of choosing what version to run.
+
+On the other hand it is also a common concept in docker and on dockerhub.
+
+As an example for [kubernetesui](https://hub.docker.com/r/kubernetesui/dashboard) they have decided not to publish a latest tag but the suggested download method on dockerhub is still presented as:
+
+```
+docker pull kubernetesui/dashboard
+```
+
+Running this example:
+
+```
+$ docker pull kubernetesui/dashboard
+Using default tag: latest
+Error response from daemon: manifest for kubernetesui/dashboard:latest not found: manifest unknown: manifest unknown
+```
+
+Which when issued results in nothing being pulled because of the missing latest tag. This might seem like a sloppy mistake but probably a very concious decision as kubernetes [tends to pull the images every time the images are used](https://kubernetes.io/docs/concepts/containers/images/#updating-images).
+
+In most cases for wikibase automatically getting a new version from the "latest" tag would put the instance in a state where a manual update would be required.
+
+As per previous discussions during the hike, one downside of making everything "too" easy to get up and running is that the end-users aren't fully aware what they are running or why they would need it. Therefore producing the latest tag seems like a easy thing to skip in order to avoid these kind of accidental selections and forcing the decision to be made by the user.
+
+## Decision
+
+- Do not publish a latest tag for now
+
+## Consequences
+
+- Remove any current latest / latest-base / latest-bundle tags for all images
```

Comment:
technically we don't need to remove latest-base/latest-bundle tags from the "generation 1" images right?
I agree however how the latest-xxx label on image that is several years old could be confusing over time. So removing sounds fine to me

Reply:
True, it's not so much a consequence but a decision probably.

Reply:
Not sure how relevant my opinion is here, but if my review is desired: I agree that in principle we should not offer the "latest" tag to help incautious users not ruin their production deployments. Everything should be pegged to a version on our side and therefore on the user's side.